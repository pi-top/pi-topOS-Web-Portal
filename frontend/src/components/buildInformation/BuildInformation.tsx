import React from "react";

import styles from "./BuildInformation.module.css";
import { BuildInfo } from "../../types/Build";

export type Props = {
  info?: BuildInfo;
};

export default ({ info }: Props) => {
  if (!info) {
    return null;
  }

  // TODO: perform a similar check with latest OS
  if (info.finalRepo === "sirius") {
    return (
      <div data-testid="build-info" className={styles.root}>
        pi-topOS Build Number: {info.buildNumber}<br />
        Release Date: {info.buildDate}
      </div>
    );
  }
  return (
    <div data-testid="build-info" className={styles.root}>
      {info.buildOsVersion && <>Build OS Version: {info.buildOsVersion}<br /></>}
      {info.buildName && <>Build Name: {info.buildName}<br /></>}
      {info.buildNumber && <>Build Number: {info.buildNumber}<br /></>}
      {info.buildDate && <>Build Date: {info.buildDate}<br /></>}
      {info.buildType && <>Build Type: {info.buildType}<br /></>}
      {info.buildRepo && <>Build Apt Repo: {info.buildRepo}<br /></>}
      {info.finalRepo && <>Final Apt Repo: {info.finalRepo}<br /></>}
      {info.ptOsWebPortalVersion && <>Web Portal Version: {info.ptOsWebPortalVersion}<br /></>}
      {info.hubFirmwareVersion && <>pi-top [4] Firmware: {info.hubFirmwareVersion}</>}
    </div>
  );
};
