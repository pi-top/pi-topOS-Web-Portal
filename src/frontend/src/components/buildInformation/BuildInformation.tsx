import React from "react";

import styles from "./BuildInformation.module.css";
import { BuildInfo } from "../../types/Build";

export type Props = {
  info?: BuildInfo;
};

export default ({ info }: Props) => {
  if (!info) {
    return null;
  }

  return (
    <div data-testid="build-info" className={styles.root}>
      pi-topOS Build Number : {info?.buildNumber}<br />
      Repository ID: {info?.buildRepo}<br />
      Build Date: {info?.buildDate}<br />
    </div>
  );
};
