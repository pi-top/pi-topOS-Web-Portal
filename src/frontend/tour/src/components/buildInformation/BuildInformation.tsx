import React from "react";

import styles from "./BuildInformation.module.css";
import { BuildInfo } from "../../types/Build";

export type Props = {
  info?: BuildInfo;
};

export default ({ info }: Props) => {
  if (!info) {
    return null;
  }

  if (info.finalRepo === "sirius") {
    return (
      <div data-testid="build-info" className={styles.root}>
        pi-topOS Build Number: {info.buildNumber}<br />
        Release Date: {info.buildDate}
      </div>
    );
  }

  return (
    <div data-testid="build-info" className={styles.root}>
      Build Name: {info.buildName}<br />
      Build Number: {info.buildNumber}<br />
      Build Date: {info.buildDate}<br />
      Build Apt Repo: {info.buildRepo}<br />
      Final Apt Repo: {info.finalRepo}
    </div>
  );
};
