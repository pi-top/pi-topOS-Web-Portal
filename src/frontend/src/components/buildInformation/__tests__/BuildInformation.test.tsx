import React, { ReactNode } from "react";
import { render, RenderResult } from "@testing-library/react";

import BuildInformation, { Props } from "../BuildInformation";

describe("BuildInformation", () => {
  let defaultProps: Props;
  let buildInformation: ReactNode;
  let rerender: RenderResult["rerender"];
  beforeEach(() => {
    defaultProps = {};

    ({ container: buildInformation, rerender } = render(
      <BuildInformation {...defaultProps} />
    ));
  });

  it("does not render by default", () => {
    expect(buildInformation).toMatchInlineSnapshot("<div />");
  });

  describe("when build info exists", () => {
    beforeEach(() => {
      defaultProps = {
        ...defaultProps,
        info: {
          buildName: "test-build",
          buildNumber: "1234567890",
          buildDate: "2020-01-01T00:00:00.000Z",
          buildRepo: "test-build-repo",
          finalRepo: "final-test-build-repo",
          buildHash: "094cdf6bc25b7429eb2820528f031afe",
        },
      };

      rerender(<BuildInformation {...defaultProps} />);
    });

    it("renders correctly", () => {
      expect(buildInformation).toMatchSnapshot();
    });

    describe('when final repo is "sirius"', () => {
      beforeEach(() => {
        defaultProps = {
          ...defaultProps,
          info: {
            ...defaultProps.info!,
            finalRepo: "sirius",
          },
        };

        rerender(<BuildInformation {...defaultProps} />);
      });

      it("renders correctly", () => {
        expect(buildInformation).toMatchSnapshot();
      });
    });
  });
});
