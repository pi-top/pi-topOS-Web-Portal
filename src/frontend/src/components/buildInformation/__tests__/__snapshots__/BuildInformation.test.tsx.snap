// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildInformation when build info exists renders correctly 1`] = `
<div>
  <div
    class="root"
    data-testid="build-info"
  >
    pi-topOS Build Number : 
    100
    <br />
    Repository ID: 
    experimental-pkgcld
    <br />
    Build Date: 
    2021-08-09
    <br />
  </div>
</div>
`;
